lib/Doctrine/ORM/Cache/DefaultCacheFactory.php

```diff
@@ -178,6 +181,19 @@ public function buildQueryCache(EntityManagerInterface $em, $regionName = null)
      */
     public function buildCollectionHydrator(EntityManagerInterface $em, array $mapping)
     {
+        if ($mapping['cache']) {
+            $targetPersister = $em->getUnitOfWork()->getEntityPersister($mapping['targetEntity']);
+
+            if ($targetPersister instanceof CachedPersister) {
+
+                $targetRegion = $targetPersister->getCacheRegion();
+
+                if ($targetRegion instanceof MultiGetRegion) {
+                    return new MultiGetCollectionHydrator($em);
```

Comment:
With the first version I was re-building the  multi-get region on each `loadCacheEntry` :-(  , while now I pass it as a constructor argument.

This makes it simpler and faster, thanks! :-)